### PR TITLE
[spark] Ray on spark: Support ray-client-server-port in `setup_ray_cluster` API

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Why are these changes needed?

Ray on spark: Support ray-client-server-port in `setup_ray_cluster` API.

API update:

```
def setup_ray_cluster(...):
    """
    Args:
        ....
    Returns:
        A tuple of (address, remote_connection_address)
        "address" is in format of "<ray_head_node_ip>:<port>"
        "remote_connection_address" is in format of
        "ray://<ray_head_node_ip>:<ray-client-server-port>",
        if your client runs on a machine that also hosts a Ray cluster node locally,
        you can connect to the Ray cluster via ``ray.init(address)``,
        otherwise you can connect to the Ray cluster via
        ``ray.init(remote_connection_address)``.
    """

```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — COMMENTED** [review]
Can we not support Ray Client? It's not recommended now.

File: python/ray/util/spark/cluster_init.py [hunk 1]
```diff
@@ -921,7 +937,14 @@ def setup_ray_cluster(
             Default value is 1.0, minimum value is 0
 
     Returns:
-        The address of the initiated Ray cluster on spark.
+        A tuple of (address, remote_connection_address)
```
Comment (jjyao) [inline python/ray/util/spark/cluster_init.py:940]: You need to change the type annotation of the function as well.

**Review by jjyao — APPROVED** [review]
It's a breaking change, are we ok with it?

File: python/ray/util/spark/cluster_init.py (hunk 1 above)
Comment (WeichenXu123) [reply python/ray/util/spark/cluster_init.py:940]: It's OK. We haven't announced GA for Ray on spark in our product side. :)

**Pull request merged**